ml/src/test/scala/ai/h2o/sparkling/ml/models/BinaryModelTestSuite.scala

```diff
@@ -0,0 +1,116 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ai.h2o.sparkling.ml.models
+
+import java.io.File
+
+import ai.h2o.sparkling.backend.BuildInfo
+import ai.h2o.sparkling.ml.algos.{H2OAutoML, H2OGBM, H2OGridSearch}
+import ai.h2o.sparkling.{SharedH2OTestContext, TestUtils}
+import org.apache.spark.sql.SparkSession
+import org.junit.runner.RunWith
+import org.scalatest.junit.JUnitRunner
+import org.scalatest.{FunSuite, Matchers}
+
+@RunWith(classOf[JUnitRunner])
+class BinaryModelTestSuite extends FunSuite with Matchers with SharedH2OTestContext {
+
+  override def createSparkSession(): SparkSession = sparkSession("local[*]")
+
+  private lazy val dataset = spark.read
+    .option("header", "true")
+    .option("inferSchema", "true")
+    .csv(TestUtils.locate("smalldata/prostate/prostate.csv"))
+
+  private lazy val referenceAlgo = new H2OGBM()
+    .setSeed(1)
+    .setFeaturesCols("CAPSULE", "RACE", "DPROS", "DCAPS", "PSA", "VOL", "GLEASON")
+    .setLabelCol("AGE")
+
+  private val outputPath = "ml/build/binary.model"
+
+  test("Exported binary model exists") {
+    referenceAlgo.fit(dataset)
+    referenceAlgo.getBinaryModel().write(outputPath)
+    assert(new File(outputPath).exists())
+  }
+
+  test("Fields on H2OBinaryModel are set after loading") {
+    referenceAlgo.fit(dataset)
+    referenceAlgo.getBinaryModel().write(outputPath)
+    val binaryModel = H2OBinaryModel.createFromBinary(outputPath)
+    assert(binaryModel.getBinaryModel().isDefined)
+  }
+
+  test("Binary model is available in H2O after training in SW") {
+    referenceAlgo.fit(dataset)
+    val binaryModel = referenceAlgo.getBinaryModel()
+    assert(H2OBinaryModel.binaryModelExists(binaryModel.modelId))
+  }
+
+  test("Binary model is uploaded to H2O cluster after loading") {
+    val path = this.getClass.getClassLoader.getResource("binary.model")
+    val model = H2OBinaryModel.createFromBinary(path.getFile)
+    H2OBinaryModel.binaryModelExists(model.modelId)
+  }
+
+  test("Load H2OBinaryModel trained in different H2O is not supported") {
+    val path = this.getClass.getClassLoader.getResource("old_version_binary.model")
+    val thrown = intercept[IllegalArgumentException] {
+      H2OBinaryModel.createFromBinary(path.getFile)
+    }
+    assert(thrown.getMessage == s"""
+                              | The binary model has been trained in H2O of version
+                              | 3.30.0.1 but you are currently running H2O version of ${BuildInfo.H2OVersion}.
+                              | To be able to use binary models, please make sure that Sparkling water/H2O is of the
+                              | same version as the version on which the binary model was trained.
+                              |""".stripMargin)
+  }
+
+  test("Loading Binary Model when Algo hasn't been fit throw exception") {
```

Comment:
nit: Would rename `Loading` to `Accessing` or `Acquiring`

Reply:
👍